### PR TITLE
Aggregated stats widgets

### PR DESCRIPTION
**Pull request by cbwebdevelopment**
See https://trello.com/c/suWzyusJ for details.

In addition to the new stat widgets display and sidebar area, this PR includes a major sweep of NPM package updates

The npm dependancies have been updated as much as as possible (i.e. React only bumped to latest 15.x, and react-router is being kept at 2.7.0). 

The dev dependancies are all updated to use the latest version (i.e. webpack, babel, test-runners, etc).

Related PRs
---
tidepool-org/tideline#357
tidepool-org/viz#120

### CODE REVIEW

**Comment by cbwebdevelopment** [comment]
[Standardize Aggregated Agg Stats Widgets [data viz] [general]](https://trello.com/c/suWzyusJ/441-standardize-aggregated-agg-stats-widgets-data-viz-general)

**Comment by cbwebdevelopment** [comment]
@krystophv I've finished up with the remaining work on this. This includes:

1. Adding the TDD/Weight stat to the basics view
2. A fair bit of styling cleanup for the new wider/responsive layout
3. A small highlighting bugfix that I noticed while cleaning up the layout of the care team page
4. Fixing a bgSource bug that was happening when scrolling back in the trends view.

The new commits start at `5b37c2d`

File: app/components/chart/basics.js [hunk 1]
```diff
@@ -1,41 +1,29 @@
-
-/*
- * == BSD2 LICENSE ==
```
Comment (krystophv) [inline app/components/chart/basics.js]: Is there a reason we dropped the license here?

Comment (cbwebdevelopment) [reply app/components/chart/basics.js]: No strong reason, other than it seemed to be excessive to me to have it at the top of each and every file.  I asked Howard a while back if there was a regulatory reason for us having it at the top of each file, and he said that having a single LICENSE file per repo is sufficient.

I've generally stopped putting this in newly added files, and figured I'd strip it out as I migrated the files over to ES6.

I'll look into this, though.  Perhaps including it is still a good practice for some reason I'm not aware of.  What's your preference?

Comment (krystophv) [reply app/components/chart/basics.js]: It's always seemed a bit overkill to have it in every file to me. I don't know that I have a strong preference either way, it just stood out to me in the diff for this file. I'd generally support moving to a single `LICENSE` file.

File: webpack.config.js [hunk 2]
```diff
@@ -1,36 +1,145 @@
...
-  __LATEST_TERMS__: JSON.stringify(process.env.LATEST_TERMS || null),
-  __PASSWORD_MIN_LENGTH__: JSON.stringify(process.env.PASSWORD_MIN_LENGTH || null),
-  __PASSWORD_MAX_LENGTH__: JSON.stringify(process.env.PASSWORD_MAX_LENGTH || null),
-  __ABOUT_MAX_LENGTH__: JSON.stringify(process.env.ABOUT_MAX_LENGTH || null),
-  __I18N_ENABLED__: JSON.stringify(process.env.I18N_ENABLED || false),
-  __DEV__: isDev,
-  __TEST__: false,
-  __DEV_TOOLS__: (process.env.DEV_TOOLS != null) ? process.env.DEV_TOOLS : (isDev ? true : false) //eslint-disable-line eqeqeq
-});
+  // {
```
Comment (krystophv) [inline webpack.config.js]: Is this comment block useful to leave in?

Comment (cbwebdevelopment) [reply webpack.config.js]: No, I just missed cleaning it up once I got the new config working.  Thanks 👍

File: app/components/chart/basics.js (hunk 1 above)
Comment (cbwebdevelopment) [reply app/components/chart/basics.js]: Cool.  I don't feel strongly either, but my slight preference is to not have them.  I'll leave this as-is, and update the copyright date in the `LICENCE` file

**Pull request merged**